### PR TITLE
auto billing

### PR DESCRIPTION
**Pull request by stsma**

### CODE REVIEW

File: TriggeredEmailer/Services/BillingService.cs [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+        {
+            //1 get starting date of billing last Week
+            var lastSunday = _sunday.AddDays(-7);
+            var saturdayDate = lastSunday.AddMinutes(_daysInterval);
+
+            //2 get all sessions from vwsession
+            var vwSessions = await _vwSessionsService.GetAll(lastSunday, saturdayDate);
+
+            //3 list of valid session per therapist
+            var validSessions = new List<KeyValuePair<int, List<vwSession>>>();
```
Comment (jshprz) [inline TriggeredEmailer/Services/BillingService.cs:50]: Nit pick: 
suggestion for simplicity: `new Dictionary<int, List<vwSession>>();`

**Review by jshprz — APPROVED** [review]
Very neat and concise code implementation! Well done, @stsma!

File: TriggeredEmailer/Models/vwSession.cs [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
+        public int sessID { get; set; }
+        public int? TherapistID { get; set; }
+        public DateTime? sessdate { get; set; }
+        public DateTime? AffirmedDate { get; set; }
+        public int? SessionStatus { get; set; }
+        public int? studentID { get; set; }
+        public string? FullName { get; set; }
+    }
+}
```
Comment (GershonHerczeg) [inline TriggeredEmailer/Models/vwSession.cs:26]: **praise:** ‏Nice use of namespaces

File: TriggeredEmailer/Services/FileLoggerService.cs [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+            var sb = new StringBuilder();
+
+            sb.AppendLine($"\r\nLog Entry : {_logLevel}");
+            sb.AppendLine($"{DateTime.Now.ToLongTimeString()} {DateTime.Now.ToLongDateString()}");
+            sb.AppendLine($"  :{_message}");
+            sb.AppendLine("-------------------------------");
+
+            await sw.WriteLineAsync(sb.ToString());
+        }
+    }
```
Comment (GershonHerczeg) [inline TriggeredEmailer/Services/FileLoggerService.cs:94]: **praise:** ‏Great use of `async / await`

File: TriggeredEmailer/Services/VWSessionsService.cs [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+        /// <param name="toDate">end date</param>
+        /// <returns></returns>
+        public async Task<ICollection<vwSession>> GetAll(DateTime fromDate, DateTime toDate)
+        {
+
+            var pSundayDate = new SqlParameter("@StartDate", SqlDbType.VarChar);
+            var pSaturdayDate = new SqlParameter("@EndDate", SqlDbType.VarChar);
+
+            pSundayDate.Value = fromDate.Date;
+            pSaturdayDate.Value = toDate.Date;
```
Comment (GershonHerczeg) [inline TriggeredEmailer/Services/VWSessionsService.cs:35]: **suggestion:** ‏Add some date validation, to make sure that values that are passed as proper date values

File: TriggeredEmailer/Services/VWStaffService.cs [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+        /// <summary>
+        /// Get all staff from VWStaff
+        /// </summary>
+        /// <returns></returns>
+        /// <exception cref="NotImplementedException"></exception>
+        public async Task<ICollection<Staff>> GetAll()
+        {
+            //TODO: 1. Map db objects to entity in vwStaff
+            //TODO: 2. Change the query to this
```
Comment (GershonHerczeg) [inline TriggeredEmailer/Services/VWStaffService.cs:28]: **praise:** ‏Great job adding TODO comments just add some future tasks to track this tech debt.

**Review by GershonHerczeg — COMMENTED** [review]
Some minor comments. Fantastic work overall! 🥇

File: TriggeredEmailer/Services/BillingService.cs (hunk 1 above)
Comment (stsma) [reply TriggeredEmailer/Services/BillingService.cs]: done @jshprz , thanks for this.

**Review by GershonHerczeg — APPROVED** [review]
🥇 great work @stsma

**Review by ttenenb — APPROVED** [review]
Great Job @stsma

**Pull request merged**